server/athenian/api/experiments/preloading/entries.py

```diff
@@ -116,6 +119,148 @@ class PreloadedReleaseToPullRequestMapper(ReleaseToPullRequestMapper):
+            if time_to is not None:
+                mask &= df["pr_created_at"] < time_to
+            if repos is not None:
+                mask &= df["repository_full_name"].isin(repos)
+
+            if len(participants) > 0:
+                mask &= cls._build_participants_mask(df, participants)
+
+            if exclude_inactive:
+                with sentry_sdk.start_span(
+                    op="_load_precomputed_done_filters/mask_creation/activity_days"
+                ):
+                    activity_days = np.concatenate(df["activity_days"].values)
+                    activity_mask = np.full(len(df["activity_days"]), False)
+                    activity_days_in_range = (
+                        time_from.replace(tzinfo=timezone.utc) <= activity_days
+                    ) & (activity_days < time_to.replace(tzinfo=timezone.utc))
+                    activity_offsets = np.cumsum(df["activity_days"].apply(len).values)
+                    indexes = np.searchsorted(
+                        activity_offsets,
+                        np.nonzero(activity_days_in_range)[0],
+                        side="right",
+                    )
+                    activity_mask[indexes] = 1
+
+                    mask &= activity_mask
+
+            match_groups, event_repos, _ = group_repos_by_release_match(
+                repos, default_branches, release_settings)
+            match_groups[ReleaseMatch.rejected] = match_groups[
+                ReleaseMatch.force_push_drop
+            ] = {"": repos}
+            if event_repos:
+                match_groups[ReleaseMatch.event] = {"": event_repos}
+
+            mask &= cls._match_groups_to_mask(df, match_groups)
+
+        done_pr_facts = cached_df.filter(
+            mask, columns=["pr_node_id", "repository_full_name", "release_match"] + [
+                c.key for c in columns],
+        )
+
+        with sentry_sdk.start_span(op="_load_precomputed_done_filters/triage"):
+            result = {}
+            ambiguous = {ReleaseMatch.tag.name: {}, ReleaseMatch.branch.name: {}}
+
+            # TODO: this `to_dict` call can probably be removed. The reason why it's here
+            # is because `dump` values requires to be `dict`, but iterating over the numpy
+            # arrays and doing the following:
+            #
+            #     dump[pr_node_id] = dict(done_pr_facts.iloc[i])
+            #
+            # is super slow because of the `dict` conversion.
+            i = 0
+            for row in done_pr_facts.to_dict(orient="records"):
+                dump = triage_by_release_match(
+                    row["repository_full_name"],
+                    row["release_match"],
+                    release_settings,
+                    default_branches,
+                    result,
+                    ambiguous,
+                )
+                if dump is None:
+                    continue
+                dump[row["pr_node_id"]] = row
+                i += 1
```

Comment:
```suggestion
            for repository_full_name, release_match, pr_node_id in zip(done_pr_facts[DonePullRequestFacts.repository_full_name.key].values, done_pr_facts[DonePullRequestFacts.release_match.key].values, done_pr_facts[DonePullRequestFacts.pr_node_id].values):
```
This is both the fastest and avoids hardcoding column names.

Reply:
I initially did this, but since `dump` values should be dictionaries of the whole row, it seemed that building the `dict` in the loop was taking a lot of time, that's why I came up with `to_dict(orient="records")`.

Reply:
Damn. OK, please leave a comment that explains this.

Reply:
It's here: https://github.com/athenianco/athenian-api/pull/1454/files#diff-0ad06f21667ae6b7fa99290cd35ebd84fdcab18d79d7711e6c505449bb328e9fR209 😄

Reply:
Changed the comment.